src/spatialdata/_core/operations/aggregate.py

```diff
@@ -1,148 +1,256 @@
 from spatial_image import SpatialImage
 from xrspatial import zonal_stats
 
 from spatialdata._core.operations.transform import transform
 from spatialdata._core.query._utils import circles_to_polygons
+from spatialdata._core.query.relational_query import get_values
 from spatialdata._types import ArrayLike
 from spatialdata.models import (
     Image2DModel,
     Labels2DModel,
     PointsModel,
     ShapesModel,
+    SpatialElement,
+    TableModel,
     get_model,
 )
 from spatialdata.transformations import BaseTransformation, Identity, get_transformation
 
+if TYPE_CHECKING:
+    from spatialdata import SpatialData
+
 __all__ = ["aggregate"]
 
 
+def _parse_element(element: str | SpatialElement, sdata: SpatialData | None, str_for_exception: str) -> SpatialElement:
+    if not ((sdata is not None and isinstance(element, str)) ^ (not isinstance(element, str))):
+        raise ValueError(
+            f"To specify the {str_for_exception!r} spatial element element, please do one of the following: "
+            f"- either pass a SpatialElement to the {str_for_exception!r} parameter (and keep "
+            f"`{str_for_exception}_sdata` = None);"
+            f"- either `{str_for_exception}_sdata` needs to be a SpatialData object, and {str_for_exception!r} needs "
+            f"to be the string name of the element."
+        )
+    if sdata is not None:
+        assert isinstance(element, str)
+        return sdata[element]
+    assert element is not None
+    return element
+
+
 def aggregate(
-    values: ddf.DataFrame | gpd.GeoDataFrame | SpatialImage | MultiscaleSpatialImage,
-    by: gpd.GeoDataFrame | SpatialImage | MultiscaleSpatialImage,
-    id_key: str | None = None,
-    *,
-    value_key: str | None = None,
-    agg_func: str | list[str] = "mean",
+    values_sdata: SpatialData | None = None,
+    values: ddf.DataFrame | gpd.GeoDataFrame | SpatialImage | MultiscaleSpatialImage | str | None = None,
+    by_sdata: SpatialData | None = None,
+    by: gpd.GeoDataFrame | SpatialImage | MultiscaleSpatialImage | str | None = None,
+    value_key: list[str] | str | None = None,
+    agg_func: str | list[str] = "sum",
     target_coordinate_system: str = "global",
+    fractions: bool = False,
+    region_key: str = "region",
+    instance_key: str = "instance_id",
+    deepcopy: bool = True,
     **kwargs: Any,
-) -> ad.AnnData:
+) -> SpatialData:
     """
     Aggregate values by given region.
 
     Parameters
     ----------
+    values_sdata
+        SpatialData object containing the values to aggregate: if None, values must be a SpatialElement; if not None,
+        values must be a string.
     values
-        Values to aggregate.
+        The values to aggregate: if values_sdata is None, must be a SpatialElement, otherwise must be a string
+        specifying the name of the SpatialElement in values_sdata
+    by_sdata
+        Regions to aggregate by: if None, by must be a SpatialElement; if not None, by must be a string.
```

Comment:
please use `by` when referring to the argument otherwise it's the same as literal by

Reply:
Done, also for `values`